content/docs/operating/security.md

```diff
@@ -0,0 +1,141 @@
+
+There are also exporters such as the SNMP and Blackbox exporters that take
+their targets from URL parameters. Thus anyone with HTTP access to these
+exporters can make them send requests to arbitrary endpoints. As they also
+support client-side authentication, this could lead to a leak of secrets such
+as HTTP Basic Auth passwords or SNMP community strings. Challenge-response
+authentication mechanisms such as TLS are not affected by this.
+
+## Client Libraries
+
+Client libaries are intended to be included in users' applications.
+
+If using a client libary provided HTTP handler, it should not be possible for
+malicious requests that reach that handler to cause issues beyond those
+resulting from additional load and failed scrapes.
+
+## Authentication/Authorisation/Encryption
+
+Prometheus and its components do not provide any server-side
+authentication, authorisation or encryption. If you require this, it is
+recommended to use a reverse proxy.
+
+Various Prometheus components support client-side authentication and
+encryption. If TLS client support is offered, there is often also an option
+called `insecure_skip_verify` which skips SSL verification.
+
+## Secrets
+
+Non-secret information or fields may be available via the HTTP API and/or logs.
+
+In Prometheus, metadata retrieved from service discovery is not considered
+secret. Throughout the Prometheus system, metrics are not considered secret.
+
+Fields containing secrets in configuration files (marked explicitly as such in
+the documentation) will not be exposed in logs or via the HTTP API. Secrets
+should not be placed in other configuration fields, as it is common for
+components to expose their configuration over their HTTP endpoint.
+
+Secrets from other sources used by dependencies (e.g. the `AWS_SECRET_KEY`
+enviroment vabiable as used by EC2 service discovery) may end up exposed due to
+code outside of our control or due to functionality that happens to expose
+wherever it is stored.
+
+## Denial of Service
+
+While there are some mitigations in place for excess load, if too many
+queries/metrics are provided components will fall over. It is more likely that
+a component will be accidentally taken out by a trusted user than malicious
+action.
+
+It is the responsibility of the the user to ensure they provide components with
+sufficient resources including CPU, RAM, disk space, iops, file descriptors,
+and bandwidth.
+
+It is recommended to monitor all components for failure, and to have them
+automatically restart on failure.
+
+## Libraries
+
+This document considers vanilla binaries built from the stock source code.
+Information presented here does not apply if you modify Prometheus source code,
+or use Prometheus internals (beyond the official client library APIs) in your
+own code.
+
+## Build Process
+
+The build pipeline for Prometheus runs on 3rd party providers to which many
+members of the Prometheus development team has access. If you are concerned
```

Comment:
access, but the real control lies with the 3rd parties which are totally independent. This should be worded more strongly.